docs/kbs.yaml

```diff
@@ -0,0 +1,267 @@
+openapi: 3.0.1
+info:
+  title: Key Broker Service (KBS) API
+  description: HTTP based API for describing the communication between a a
+               Key Broker Client in a confidential guest and a trusted
+               Key Broker Service.
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0
+
+servers:
+- url: http://<kbs>/kbs/v0
+
+paths:
+
+  /auth:
+    post:
+      summary: Initiate an attestation authentication
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Request'
+
+      responses:
+        200:
+          description: The attestation challenge
+          content:
+            headers:
+              Set-Cookie:
+                name: kbs-session-id
+                schema:
+                  type: string
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Challenge'
+        400:
+          description: The request content or format is not supported
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorInformation'
+
+  /attest:
+    post:
+      summary: Send an attestation evidence
+      parameters:
+        in: cookie
+        name: kbs-session-id
+        schema:
+          type: string
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Attestation'
+
+      responses:
+        200:
+          description: The attestation evidence is authenticated
+        401:
+          description: Missing or invalid session ID
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorInformation'
+
+  /resources/key/<key_id>/:
+    get:
+      summary: Get a key from the Key Broker Service
+      parameters:
+        in: cookie
+        name: kbs-session-id
+        schema:
+          type: string
+      responses:
+        200:
+          description: The Key Broker Service reponse that includes the key
+          content:
+            headers:
+              Set-Cookie:
+                name: kbs-session-id
+                schema:
+                  type: string
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponseHeader'
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ResponsePayload'
```

Comment:
This looks like it's defining two response bodies.  I don't think this is what you mean :-)  Also, syntactically, I guess this would not validate against the OpenAPI schema.  BTW, have you tried this against one of the [validation tools](https://openapi.tools/#description-validators)?  As a recommendation, I'd hook one in (e.g., via Makefile) to check both as you develop and as part of the CI.

Reply:
@thomas-fossati I wonder what would be the proper way to represent a JWT or a JWE with an OpenAPI schema.

Using the Compact Serialization scheme, iiuc it would be something like:

```yaml
Response:
     required:
          - jwe
        type: object
        properties:
          jwe: string
          description: A JWE serialized with Compact Serialization
```

A less compact option would be to use the Flattened Serialization syntax, which would make it look like that:

```yaml
Response:
      required:
        - unprotected
        - ciphertext
        - encrypted_key
      type: object
      properties:
        unprotected:
          type: string
        ciphertext:
          type: string
        encrypted_key:
          type: string
      description: JWE serialized with the Flattened Serialization format
```

AFAICT there are no OpenAPI standard ways of defining a JWE or JWS.

Reply:
@thomas-fossati Thanks for the offline advice (https://json-schema.org/draft/2020-12/json-schema-validation.html#section-8.6).
I think the OpenAPI 3.1 description looks good now, and it's validated with the redocly linter.

Reply:
👍 awesome job!